tools/spring/spring-oauth2-utilities/src/main/java/org/opfab/springtools/configuration/oauth/UpdateUserListener.java

```diff
@@ -47,12 +48,29 @@ public UpdateUserListener( AmqpAdmin amqpAdmin, FanoutExchange userExchange, Con
      * <p>Constructs a non durable queue to exchange using queue name</p>
      * @return
      */
-    private Queue createQueue(String queueName, FanoutExchange exchange) {
+    private Queue createQueue(String queueName, FanoutExchange exchange, int retries, long retryInterval) {
         Queue queue = QueueBuilder.nonDurable(queueName).build();
-        amqpAdmin.declareQueue(queue);
-
-        Binding binding = BindingBuilder.bind(queue).to(exchange);
-        amqpAdmin.declareBinding(binding);
+        boolean created = false;
+        int retry = 1;
+        while (!created) {
+            try {
+                amqpAdmin.declareQueue(queue);
+    
+                Binding binding = BindingBuilder.bind(queue).to(exchange);
+                amqpAdmin.declareBinding(binding);
+                created = true;
+            } catch (AmqpException ex) {
+                if (retry == retries) {
+                    throw ex;
+                }
+                try {
+                    Thread.sleep(retryInterval);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+                retry++;
+            }
+        }
```

Comment:
To avoid duplication , add this method in a util class in tools/generic/utilities/src/main/java/org/opfab/utilities/amqp.java

Reply:
Done